src/Combine.elm

```diff
@@ -239,21 +239,26 @@ maybeIgnore (Parser p) =
 
 
 many : Parser state a -> Parser state (List a)
-many p =
-    manyWithoutReverse p
-        |> map List.reverse
+many (Parser p) =
+    let
+        manyWithoutReverseStep : ( state, List a ) -> Core.Parser (Core.Step ( state, List a ) ( state, List a ))
+        manyWithoutReverseStep ( oldState, items ) =
+            Core.oneOf
+                [ p oldState
+                    |> Core.map (\( newState, item ) -> Core.Loop ( newState, item :: items ))
+                , Core.succeed (Core.Done ( oldState, List.reverse items ))
```

Comment:
I think this should be wrapped in lazy so that we don't compute `List.reverse items` when there are more items left.

Reply:
indeed, fixed. I do wonder why this didn't result in a bigger penalty in performance :thinking: